spring-core/src/main/java/org/springframework/util/FastByteArrayOutputStream.java

```diff
@@ -504,13 +500,10 @@ public void updateMessageDigest(MessageDigest messageDigest) {
 		 */
 		@Override
 		public void updateMessageDigest(MessageDigest messageDigest, int len) {
-			if (this.currentBuffer == null) {
+			if (this.currentBuffer == null || len == 0) {
```

Comment:
Please revert the changes to this method, since the subsequent inline comment does not apply to the `len == 0` case.